### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by devennorton**
Its all new. I would not bother too much going commit by commit.  I would look over the tests cause they're short. and I woudl start with the SumoBuildNotifier, and probably look at BuildModelFactor.  the resources will be pretty opaque to anyone not familiar with jenkins templating and are trivial besides.  

I'll add some more notes in a bit

@jakozaur @cddude229 please peruse at your pleasure

### CODE REVIEW

File: src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/BuildModelFactory.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+      for (MavenBuild module : modules.values()) {
+        mBuildModel.addModule((MavenModuleBuildModel) generateBuildModelFor(module));
+      }
+      buildModel = mBuildModel;
+    } else if (build instanceof MavenBuild) {
+      MavenBuild mbuild = (MavenBuild) build;
+      MavenModuleBuildModel mBuildModel = new MavenModuleBuildModel();
+      BuildModelFactory.populateGeneric(mBuildModel, mbuild);
+/*      for(SuiteResult result: mbuild.getAction(SurefireReport.class).getResult().getSuites()) {
```
Comment (cddude229) [inline src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/BuildModelFactory.java]: Don't check in commented code.  (One more line above.)

**Comment by cddude229** [comment]
lgtm.  Leaving for Jacek to review.

File: src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/BuildModelFactory.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import hudson.maven.MavenModuleSetBuild;
+import hudson.maven.reporters.SurefireAggregatedReport;
+import hudson.model.AbstractBuild;
+import hudson.tasks.test.TestResult;
+
+import java.util.Map;
+
+
+/**
+ * Created by deven on 7/11/15.
```
Comment (jakozaur) [inline src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/BuildModelFactory.java]: Please add one line explaining what this class do.

File: src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/SumoBuildNotifier.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.commons.httpclient.methods.StringRequestEntity;
+import org.apache.http.protocol.HTTP;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Created by deven on 7/6/15.
```
Comment (jakozaur) [inline src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/SumoBuildNotifier.java]: Would also add one line what it does.

**Comment by jakozaur** [comment]
lgtm,

would only add a few one-line comments and README file explaining what this plugin do.

**Pull request closed without merging**